R/get_smr.R

```diff
@@ -1,96 +1,165 @@
-#' Perform indirect rate standardization
+#' Calculate a standardized morbidity/mortality ratio
 #'
-#' @description Compute a standardized mortality ratio (SMR) or expected number
-#' of deaths given the number of observed deaths, the size of each stratum in the
-#' observed population, and the specific rate for each stratum in the standard
-#' population.
+#' @description Calculate a standardized morbidity/mortality ratio (SMR) or the
+#' expected number of events in each stratum. The SMR is the number of observed
+#' events divided by the expected events.
 #'
-#' @param count A scalar, the number of observed deaths.
-#' @param popn A numeric vector containing the size of each defined population.
-#' @param std_rt A numeric vector containing the specific rate (expressed as
-#' the number of cases per unit of population) for each stratum in the standard
+#' @param count A scalar, the number of observed events.
+#' @param popn A numeric vector containing the size of each stratum in the observed
 #' population.
-#' @param output_type If `"ratio"` (default), only the SMR is returned. If `"counts"`,
-#' the expected number of deaths for each stratum in the observed population is
-#' returned as a vector.
-#' @param percent If `FALSE` (default), the SMR is returned as is (e.g. 0.5). If `TRUE`, the
-#' SMR is expressed as a percentage (e.g. 50).
+#' @param std_r A numeric vector containing the rate or risk of each stratum in
+#' the standard population.
+#' @param std_measure A string to indicate whether the values in `std_r` are rates
+#' or risks, which is used to calculate the expected number of events in each stratum.
+#' Currently, this function only supports the calculation of rate SMRs (default is `"rate"`).
+#' The calculation of risk SMRs will be implemented in the future.
+#' @param output_type If `"ratio"` (default), the SMR is returned. If `"counts"`,
+#' the expected number of events in each stratum is returned.
+#' @param percent If `FALSE` (default), the SMR is returned as is (e.g. 0.5). If
```

Comment:
While it might be more formal to explicitly state the data type of each parameter, I did a bit of digging to see if it's "acceptable" to just describe the expected behavior of each valid argument. 

Below are some screenshots from the `remotes::install_github` and `tidy::pivot_longer` docs:

![image](https://user-images.githubusercontent.com/54685414/147302973-28535625-dde8-40e8-8ea0-2c850724e69f.png)

![image](https://user-images.githubusercontent.com/54685414/147303027-600a4945-2311-4d02-95f1-a95279dd6a83.png)

If these well-respected packages write their docs in this way, hopefully it's alright if we keep ours the way it is?

Reply:
Following other R packages, I don't think we need to be explicit about the type when the type is implicit and especially if we don't check for a specific type. I wrote this in another conversation but using "If `TRUE`" is different from If "TRUE" and the first case makes it clear that this is something that should evaluate to `TRUE`. Also, technically, the input is not restricted to a Boolean object, the user can put in a statement that evaluates to `TRUE` or `FALSE` and it will still work (as intended).